### PR TITLE
Open the bag directory instead of a single file

### PR DESCRIPTION
**Pull request by mjeronimo**
In ROS2, there is no longer a single rosbag file. Let the directory
containing the various files be used as the "rosbag." Also, use a
consistent proposed filename format for both the Record and Save dialogs.
Finally, fix a minor issue in the raw view.

Signed-off-by: Michael Jeronimo <michael.jeronimo@openrobotics.org>

### CODE REVIEW

File: rqt_bag/src/rqt_bag/plugins/raw_view.py [hunk 1]
```diff
@@ -208,7 +208,7 @@ def _add_msg_object(self, parent, path, name, obj, obj_type):
 
         elif type(obj) in [str, bool, int, long, float, complex, Time]:
             # Ignore any binary data
-            obj_repr = codecs.utf_8_decode(str(obj).decode(), 'ignore')[0]
+            obj_repr = codecs.utf_8_decode(str(obj).encode(), 'ignore')[0]
```
Comment (jacobperron) [inline rqt_bag/src/rqt_bag/plugins/raw_view.py:211]: This change looks unrelated. Should it be included in this PR?

**Review by jacobperron — COMMENTED** [review]
LGTM, one question.

**Comment by mjeronimo** [comment]
Yeah, doesn't have to be. I didn't know if it was worth a separate PR for a single line.

File: rqt_bag/src/rqt_bag/plugins/raw_view.py (hunk 1 above)
Comment (mjeronimo) [reply rqt_bag/src/rqt_bag/plugins/raw_view.py:211]: Yeah, doesn't have to be. I didn't know if it was worth a separate PR for a single line.

Comment (jacobperron) [reply rqt_bag/src/rqt_bag/plugins/raw_view.py:211]: In general, I would try to keep changes focused on one thing for traceability (e.g. when reviewing commit history). I find it makes it easier to pinpoint where and why something was changed. It can also make porting/reverting specific changes easier when they are more focused.

In this case, we can leave it as-is since I see `encode()` was added in ROS 1 in #68, and I guess `decode()` was added accidentally during the port in https://github.com/ros-visualization/rqt_bag/pull/72 (independent of the ROS 1 change).

**Review by jacobperron — APPROVED** [review]

File: rqt_bag/src/rqt_bag/plugins/raw_view.py (hunk 1 above)
Comment (mjeronimo) [reply rqt_bag/src/rqt_bag/plugins/raw_view.py:211]: Ok. Yes, that's correct. Fixing an issue introduced accidentally.
